gridData/OpenDX.py

```diff
@@ -611,10 +611,8 @@ class DXParser(object):
     |"(?P<QUOTEDSTRING>[^\"]*)"   # string in double quotes  (quotes removed)
     |(?P<WHITESPACE>\s+)          # white space
     |(?P<REAL>[-+]?               # true real number (decimal point or
-          (\d+\.\d*([eE][-+]\d+)?)  # scientific notation)
-          |(\d*\.\d+([eE][-+]\d+)?)
-          |(\d[eE][-+]\d+))
-    |(?P<INTEGER>[-+]?\d+)       # integer
```

Comment:
I am not thrilled throwing out the INTEGER pattern. Could you add a test case with an example that fails with the original code, something that contains
```
0.0     -1e-06  0.0     
```
Maybe just update [gridData/tests/test.dx](https://github.com/MDAnalysis/GridDataFormats/blob/master/gridData/tests/test.dx) with some offending lines or add a new minimal tests file.

Basically, we want to have a test that shows that the bug has been squashed.

Reply:
It should be possible to fix the regular expressions... I probably made a subtle mistake when I wrote them a long time ago.

Reply:
I modified `test.dx` in `0fabaa9`and `3598ae5`.  I don't know what framework you are using for your test suite or how to run the tests.  I am really only familiar with the `unittest` module. I can confirm that this test file now loads correctly but would fail at multiple points in the previous version.

I tried to keep the `INTEGER` pattern but anything that is an integer is captured by `FLOAT` since regular expressions are greedy.  I only deleted it because it didn't do anything after I updated the `FLOAT` expression. It might be possible to get `INTEGER` to work by looking ahead or back but I didn't investigate this. BTW, I just took the `FLOAT` regex from the [`re`](https://docs.python.org/3/library/re.html#simulating-scanf) documentation.

Reply:
Ok, thanks for the explanation. Then let's go with your modifications and leave out INTEGER.

Can you add a comment to the regexes that references the source as the re module with the link that you gave and that states that FLOAT will also catch integers?

Reply:
Done.